client/XQDebugger/xqyDebugConfigProvider.ts

```diff
@@ -106,11 +106,17 @@ export class XqyDebugConfigurationProvider implements DebugConfigurationProvider
             }
         }
 
-        if (!config.program) {
-            config.program = window.activeTextEditor.document.fileName;
-            config.query = window.activeTextEditor.document.getText();
-        } else {
-            config.query = readFileSync(config.program).toString();
+        // Need to tell VSCode what parameters to use for "launching" a debug session based on a configuration in launch.json
+        // Note that this information is unnecessary for "attaching" to a remote request for debugging.
+        if (config.request === 'launch') {
+            if (!config.program) {
```

Comment:
Why does this conditional differ based on whether it's XQuery or JS? JS doesn't have an `else` condition, for example. I was initially thinking that it would be a helper function with common logic for both - i.e. "If no program is specified, then use the current editor".

Reply:
My guess is that the "else" clause was added to the xqy version when somebody wanted to debug a specific module without having that module active (maybe they were editing a library module).
The same requirement probably just never popped up in JS.
I'm happy to add this to the JS version if you think I should.

Reply:
Added a new story to clean this up a bit and to make it clearer to the user.
https://project.marklogic.com/jira/browse/DEVEXP-403